core/2d/CCDrawNode.cpp

```diff
@@ -369,8 +369,9 @@ void DrawNode::drawCircle(const Vec2& center,
     }
     if (_lineWidth > minThickness)
     {
-        auto scaleFactor = AX_CONTENT_SCALE_FACTOR() * 2 ;
-        drawPolygon(vertices, segments, Color4B(1.0f, 0.0f, 0.0f, 1.0f), _lineWidth / scaleFactor, color);
+        GLView* glView = _director->getOpenGLView();
+        auto borderScale = (glView->getScaleX() + glView->getScaleY());
+        drawPolygon(vertices, segments, Color4B(1.0f, 0.0f, 0.0f, 1.0f), _lineWidth / borderScale , color);
```

Comment:
```suggestion
        drawPolygon(vertices, segments, Color4B(1.0f, 0.0f, 0.0f, 1.0f), _lineWidth * borderScale , color);
```

Reply:
With `_lineWidth * borderScale`i getting this result:

![image](https://user-images.githubusercontent.com/8652787/218058233-c1f5ad7c-439c-49bc-953a-f211900dc78b.png)

Reply:
Because the resourceSize is 960x640, and designSize is 480, 320 then the scale is 2.0, it's correct behavior

Reply:
The green ring should have the same behavior like the red ring but without the rendering issue.
The green ring "is the red ring" without the bad rendering issue.

Reply:
@halx99 
You anderstand what I mean?
The green is only needed to make a better rendering if the _lineWidth to big and normaly the "red circle show renderuing issues.

Reply:
If I set both resourceSize and designSize to 1280x720, and limit the `thick` < 10, then looks like:
![image](https://user-images.githubusercontent.com/6977319/218068445-6dffc1e2-8792-47ce-9de9-a341125b23ab.png)

Reply:
And 960x640, 480x320 looks like:
![image](https://user-images.githubusercontent.com/6977319/218068952-6cd279f4-75ed-43fe-8451-4503d700ff2f.png)

Reply:
The green api is only a workaourond if the red api not rendering as the developer want (in most cases the thick will be "1-2' pixel)

Reply:
Both 960x640:
![image](https://user-images.githubusercontent.com/6977319/218071070-15164174-77f6-482d-8e30-2ff8ec82e1e9.png)

All my screeshot use follow code:
```cpp
void DrawNode::drawCircle(const Vec2& center,
                          float radius,
                          float angle,
                          unsigned int segments,
                          bool drawLineToCenter,
                          float scaleX,
                          float scaleY,
                          const Color4B& color,
                          float minThickness)
{
    const float coef = 2.0f * (float)M_PI / segments;

    auto vertices = _abuf.get<Vec2>(segments + 2);

    auto scaleFactor = AX_CONTENT_SCALE_FACTOR();

    for (unsigned int i = 0; i <= segments; i++)
    {
        float rads    = i * coef;
        vertices[i].x = radius * cosf(rads + angle) * scaleX * scaleFactor + center.x;
        vertices[i].y = radius * sinf(rads + angle) * scaleY * scaleFactor + center.y;
    }
    if (_lineWidth > minThickness)
    {
        GLView* glView = _director->getOpenGLView();
        auto borderScale = (glView->getScaleX() + glView->getScaleX()) / 2.0f;
        drawPolygon(vertices, segments, Color4B(0.0f, 0.0f, 0.0f, 0.0f), _lineWidth * borderScale, color);
    }
    else
    {
        if (drawLineToCenter)
        {
            vertices[segments + 1] = center;
            drawPoly(vertices, segments + 2, true, color);
        }
        else
            drawPoly(vertices, segments + 1, true, color);
    }
}
```

Reply:
> And 960x640, 480x320 looks like: ![image](https://user-images.githubusercontent.com/6977319/218068952-6cd279f4-75ed-43fe-8451-4503d700ff2f.png)

The red looks good. so the green is not needed ;)

Reply:
The example on th cpp-test using 2 as parameter maybe thats too smal better I use 4 or 5

Reply:
Try without any scale: 960x640, resizing window:
![image](https://user-images.githubusercontent.com/6977319/218071796-19b8ee97-3e74-4543-be75-e6ee08dc3e95.png)

![image](https://user-images.githubusercontent.com/6977319/218071821-d6d6826d-6b72-4113-a61e-eb92005ee578.png)

![image](https://user-images.githubusercontent.com/6977319/218071866-16cb8902-3310-42a8-91de-6b53a568e40e.png)

Reply:
code is:
```cpp
void DrawNode::drawCircle(const Vec2& center,
                          float radius,
                          float angle,
                          unsigned int segments,
                          bool drawLineToCenter,
                          float scaleX,
                          float scaleY,
                          const Color4B& color,
                          float minThickness)
{
    const float coef = 2.0f * (float)M_PI / segments;

    auto vertices = _abuf.get<Vec2>(segments + 2);

    auto scaleFactor = AX_CONTENT_SCALE_FACTOR();

    for (unsigned int i = 0; i <= segments; i++)
    {
        float rads    = i * coef;
        vertices[i].x = radius * cosf(rads + angle) * scaleX * scaleFactor + center.x;
        vertices[i].y = radius * sinf(rads + angle) * scaleY * scaleFactor + center.y;
    }
    if (_lineWidth > minThickness)
    {
        drawPolygon(vertices, segments, Color4B(0.0f, 0.0f, 0.0f, 0.0f), _lineWidth, color);
    }
    else
    {
        if (drawLineToCenter)
        {
            vertices[segments + 1] = center;
            drawPoly(vertices, segments + 2, true, color);
        }
        else
            drawPoly(vertices, segments + 1, true, color);
    }
}
```

Reply:
So, finally, remove any scale of it?

Reply:
Oh dear,
You misunderstand the use case for the green API.

Its only needed is the thick so many as the red api will be wrong rendering.
The developer have to check it and change the api (if it needed).
The example is bad for this: => "must" have to do which is part of the developer

Reply:
In your last three examples the red api is working good enough and it is faster as the green api too.

Reply:
> So, finally, remove any scale of it?

without any scale is look really often bad  (i belive)

Reply:
Last but not least: In most cases is the red api working well.
Only if the "programmwer" want a thick > 10 or so and using circles the rendering goes badly.
Thast is not a no go...here can the green api making a better "look" of the ring. thats all.

Reply:
The example is bad for "comparing" because the developer is using only the green OR the red api